src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py

```diff
@@ -80,8 +103,38 @@ def __init__(self, start: int, end: int):
         )
         self._condition = threading.Condition()
         self._no_remote_peer = 0
+        self._metadata_keys = set(key.strip().lower() for key in metadata_keys)
```

Comment:
nit: Might be better to make a method/function sanitize_metadata_key, and use it both here and in `_add_metadata`